### DISCUSSION TITLE
How to remove ".": {}, from JSON?

### DISCUSSION THREAD

**Discussion by patrickdung** [vectordotdev/vector#18720] [Q&A]
The program that generate the json produce something like this:
(extracted)
```
                          "k:{\"name\":\"HOSTNAME\"}": {
                            ".": {},
                            "f:name": {},
                            "f:valueFrom": {
                              ".": {},
                              "f:fieldRef": {}
                            }
                          },
```

I would like to remove these ```".": {},```

How to write it in VRL in Vector?
```new_message = replace(string!(.message), "\".\": {},", "")```
but it complain about the period

**Comment by jszwedko** [comment]
I would recommend parsing the message as JSON and then removing the field rather than doing string replacement as this is likely to be much more brittle. See an example here:

https://playground.vrl.dev/?state=eyJwcm9ncmFtIjoiLiA9IHBhcnNlX2pzb24hKC5tZXNzYWdlKVxuZGVsKC5cIms6e1xcXCJuYW1lXFxcIjpcXFwiSE9TVE5BTUVcXFwifVwiLlwiLlwiKSIsImV2ZW50Ijp7Im1lc3NhZ2UiOiJ7XCJrOntcXFwibmFtZVxcXCI6XFxcIkhPU1ROQU1FXFxcIn1cIjp7XCIuXCI6e30sXCJmOm5hbWVcIjp7fSxcImY6dmFsdWVGcm9tXCI6e1wiLlwiOnt9LFwiZjpmaWVsZFJlZlwiOnt9fX19In0sImlzX2pzb25sIjpmYWxzZSwiZXJyb3IiOm51bGx9

**Reply by patrickdung** [reply]
Thanks for reply..

But how about the source JSON has multiple ```".": {},```?
The text ```"k:{\"name\":\"HOSTNAME\"}": {``` could be any text and the ```".": {},``` comes after it.

**Reply by jszwedko** [reply]
I think I'd need to see another few examples since I'm not sure I'm following.
